File: codebattle/js/word_pattern.js
//Задача: word_pattern

//Given a pattern and a string str, find if str follows the same pattern. Examples pattern = " abba ", str = " dog cat cat dog " should return true.

//Example: true == solution("zzbinqs", "dragonborn dragonborn by his honor in sworm")

//https://github.com/h/b/blob/master/src/b/issues/word_pattern.clj

module.exports = (a, b) =>
{
  a = a.split("");
  b = b.split(" ");
  
  if(a.length != b.length)
    return false;
  
  var m = {};
  for(var i = 0; i < a.length; i++)
  {
    if(typeof m[a[i]] !== "string")
      m[a[i]] = b[i];
    else
      if(m[a[i]] !== b[i])
      {
        console.log(m[a[i]] + " " + b[i]);
        return false;
      }
  }

File: codebattle/python/word_pattern.py
#Задача: word_pattern

#Given a pattern and a string str, find if str follows the same pattern. Examples pattern = " abba ", str = " dog cat cat dog " should return true.

#Example: true == solution("zzbinqs", "dragonborn dragonborn by his honor in sworm")

#https://github.com/h/b/blob/master/src/b/issues/word_pattern.clj

# BEGIN implement function `solution` here 
def solution(s1,s2):
  ss2 = s2.split()
  if len(s1) != len(ss2):
    return False
  d = []
  for i in range(len(s1)):
    d.append((s1[i],ss2[i]))
  s = set(d)
  tt = [t[0] for t in s]
  return len(s) == len(tt)
  

# END